src/Microsoft.AspNet.Mvc/MvcServiceCollectionExtensions.cs

```diff
@@ -22,7 +22,7 @@ private static void ConfigureDefaultServices(IServiceCollection services, IConfi
             services.AddOptions(configuration);
             services.AddDataProtection(configuration);
             services.AddRouting(configuration);
-            services.AddContextAccessor(configuration);
+            services.AddScopeLocal(configuration);
```

Comment:
You aren't reading this correctly, this is the extension method that is adding IScopeLocal as a scoped service to DI...